enterprise/api/v1/serializers.py

```diff
@@ -278,15 +278,16 @@ class EnterpriseCatalogCoursesReadOnlySerializer(serializers.Serializer):
     previous = serializers.CharField(read_only=True, help_text=_("URL to fetch previous page of courses."))
     results = serializers.ListField(read_only=True, help_text=_("list of courses."))
 
-    def update_enterprise_courses(self, request, catalog_id):
+    def update_enterprise_courses(self, request, catalog_id, enterprise_customer=None):
         """
         This method adds enterprise specific metadata for each course.
 
         We are adding following field in all the courses.
             tpa_hint: a string for identifying Identity Provider.
         """
         courses = []
-        enterprise_customer = utils.get_enterprise_customer_for_user(request.user)
+        if enterprise_customer is None:
+            enterprise_customer = utils.get_enterprise_customer_for_user(request.user)
```

Comment:
The other courses endpoint by catalog uses the same serializer, and does not pass in enterprise_customer. I thought this would be simpler as to not change that other view. If we choose to deprecate that, then we can make this required.